scripts/jenkins/build_docker.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -euxo pipefail
+
+if ! command -v docker
+then
+  echo "ERROR: Building Docker image requires Docker binary to be installed" && exit 1
+elif ! docker version
+then
+  echo "ERROR: Building Docker image requires Docker daemon to be running" && exit 1
+elif ! command -v curl
+then
+  echo "ERROR: Building Docker image requires cURL to be installed" && exit 1
+fi
+
+echo "INFO: Fetching latest tag"
+#TODO Move this into apm-pipeline-library
```

Comment:
There are different env variable to know the tag name:
- TAG_NAME
- BRANCH_NAME
- GIT_BRANCH

Some references:
- https://apm-ci.elastic.co/pipeline-syntax/globals#env
- https://github.com/jenkins-infra/jenkins.io/blob/master/content/doc/book/pipeline/syntax.adoc#when

Reply:
I modified it to use `TAG_NAME` if it is present in the environment. Thanks!